### PR TITLE
Core: Remove deprecated AssertHelpers

### PR DESCRIPTION
**Pull request by liuxiaocs7**
- SubTask of https://github.com/apache/iceberg/issues/7094

Remove `AssertHelpers` in `iceberg-core-avro` module.

### CODE REVIEW

File: core/src/test/java/org/apache/iceberg/avro/TestReadProjection.java [hunk 1]
```diff
@@ -712,14 +713,21 @@ public void testEmptyNestedStructRequiredProjection() throws Exception {
                     Types.NestedField.required(2, "inner", Types.StructType.of()))));
 
     Record projected = writeAndRead("nested_empty_req_proj", writeSchema, emptyStruct, record);
-    AssertHelpers.assertEmptyAvroField(projected, "id");
+    assertEmptyAvroField(projected, "id");
+
```
Comment (nastra) [inline core/src/test/java/org/apache/iceberg/avro/TestReadProjection.java:717]: nit: unnecessary new line

**Review by nastra — APPROVED** [review]
the PR seems to be small enough, can we include additional changes for `iceberg-core`?

File: core/src/test/java/org/apache/iceberg/avro/TestReadProjection.java (hunk 1 above)
Comment (liuxiaocs7) [reply core/src/test/java/org/apache/iceberg/avro/TestReadProjection.java]: done!

**Review by nastra — APPROVED** [review]
LGTM, thanks @liuxiaocs7

**Pull request merged**